### DISCUSSION TITLE
p.169 SECTION-066 リスト17.4 RespondJSON関数のエラーログ出力について

### DISCUSSION THREAD

**Discussion by magavel** [budougumi0617/go_todo_app#61] [Ideas]
@budougumi0617 さん

お世話になっております。
先月のカジュアル面談ではありがとうございました！

一通り読み終えたので、気になった部分をページごとにdiscussionを作成していければと思います。
※手元の書籍は第1刷（2022年8月1日発行）です。

---

書籍ではRespondJSON関数でエラーがあった場合のログ出力に、logパッケージではなくfmtパッケージを使っていました。
logパッケージをインポートしていてサンプルコードでもlogを使っているので、誤植でしょうか？

p.169 SECTION-066 リスト17.4

```diff
package handler

import (
	"context"
	"encoding/json"
	"fmt"
	"log"
	"net/http"
)

// 省略

func RespondJSON(ctx context.Context, w http.ResponseWriter, body any, status int) {

	// 省略

		if err := json.NewEncoder(w).Encode(rsp); err != nil {
-			fmt.Printf("write error response error: %v", err)
+			log.Printf("write error response error: %v", err)
		}
		return
	}

	w.WriteHeader(status)
	if _, err := fmt.Fprintf(w, "%s", bodyBytes); err != nil {
-		fmt.Printf("write response error: %v", err)
+		log.Printf("write response error: %v", err)
	}
}
```

**Comment by budougumi0617** [comment]
ご指摘ありがとうございます。
`log` pkg使うと出力がJSONでなくなってしまうのでここは`fmt`使ったほうがよいですね。サンプルコードなどもその方向で修正いたしました。ありがとうございます！

**Reply by budougumi0617** [reply]
正誤表に反映しました。ご指摘ありがとうございました！
https://github.com/budougumi0617/go_todo_app/releases/tag/v1.0.8
